% Introduction / signal treatment
In this chapter we will present the original methods proposed in our work. First, we formalize our definition of genomic signal and describe how they are created from aligned reads for DNase-seq and histone modification ChIP-seq (Section~\ref{sec:genomic.signal}). In posession of such genomic signal perform a number of signal pre-processing steps. The first, regards corrections from experimental biases on DNase-seq data (Section~\ref{sec:experimental.bias.correction}). The second, concerns DNase-seq and histone modification ChIP-seq signal normalization and scaling in order to account for within- and between-data set variability (Section~\ref{sec:signal.treatment}).

% HMM and filters
Then, given the pre-processed signals, we proceed to the description of the original methods used here to address the problem of the identification of active transcription factor binding sites. The first method regards a hidden Markov model-based approach (Section~\ref{sec:hidden.markov.models}). The second method regards an approach based on treating the signals with eletronic filters (Section~\ref{sec:signal.processing.filters}).

% Statistical methods and conclusion
Finally, we describe all statistical techniques used in this study (Section~\ref{sec:statistical.methods}) and close the chapter by providing a brief discussion on the methodologies chosen to address the problem (Section~\ref{sec:discussion.3}).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%




