\chapter{Background}
\label{chapter_background}
We first introduce the fundamental biological concepts of DNA, gene expression and epigenetics. 
We then explain ChIP-seq, a method to investigate epigenetics by identifying DNA-protein complexes in a genome wide manner.
Next, we detail the ChIP-seq data analysis.
We introduce the peak calling problem on a single ChIP-seq signal.
The extension of this problem is the differential peak calling problem.
This problem is the target of current research which also includes this thesis.
We motivate the differential peak calling with current biological and medical studies and point out arising challenges.
Next, we discuss previous work that is related to this thesis.
Finally, we formulate the aims of the thesis.

\section{Biology}
We give an overview of the biological concepts that are necessary to understand the thesis.
See~\cite{Alberts2002} for a more detailed description of molecular biology and, in particular, for a gentle introduction to DNA. 
\cite{Lodish2007} give a detailed introduction to gene expression and \cite{Allis2007} explain epigenetics in detail.

\subsection{DNA}
Deoxyribonucleic acid (DNA) is the carrier of genetic information of living organisms.
DNA is a chain molecule with nucleotides as elements.
While the phosphate group and the sugar molecule are similar, the third element of a nucleotide, the nucleobase (or base), varies.
We therefore can describe a DNA strand by its bases adenine (\nuc{A}), cytosine (\nuc{C}), guanine (\nuc{G}) and thymine (\nuc{T}).
DNA is directional, that is, it has a 5' and a 3' end.
The nucleotides adenine and thymine as well as cytosine and guanine can pair and form a double-stranded structure.
Both strands are coiled around each other and build the typical double helix.
The strands are reverse complements of each other.


\subsection{Gene Expression}
Gene expression is the process of selectively reading genetic information contained in the DNA.
Processing the genetic information works in two steps: first, DNA is translated into RNA molecules and second, the RNA is translated into proteins.
In the first step, a protein complex called RNA polymerase II binds to the DNA and successively reads the genetic information of a DNA molecule (transcription).
Figure~\ref{pic_gene_transciption} shows the concept of gene transcription.
The RNA polymerase complex attaches to the promoter of gene X, locally separates the two DNA strands, creates an RNA molecule by reading one nucleotide at a time and finally reconnects the two DNA strands.
In the second step, a particular protein translates the RNA to protein molecules (translation).
Proteins are life-essential molecules which contribute to the structural components of a cell and perform all activities within a cell.
% In this thesis we concentrate on the transcription step of gene expression, as it is the first step to generate proteins.
Various control mechanisms of a cell facilitate production of proteins on demand.
For example, in Figure~\ref{pic_gene_transciption} certain proteins, so called transcription factors (TFs), attach to the DNA and may effect the rate of transcription initiation.

\begin{figure}[ht]
  \centering
  \includegraphics[width = 15cm]{pics/gene_transcription}
\caption[Gene transcription]{Gene transcription. 
In this example the transcription of gene X is shown. 
The RNA polymerase II attaches to the gene promoter, a DNA sequence that is upstream located to the gene.
Also, general transcription factors bind to the promoter and help the polymerase to position properly at the promoter.
Several control mechanisms determine the gene transcription.
For example, gene regulatory proteins such as TFs bind to regulatory sequences and effect the rate of transcription initiation.
Regulatory sequences, also called enhancer regions, may either be located close to the promoter, far upstream or close downstream of the gene.
The figure is based on \cite{Alberts2002}.
}
\label{pic_gene_transciption}
\end{figure}

\subsection{Epigenetics}
\label{sec_epigenetics}
Epigenetics investigates changes in gene expression by mechanisms other than variation in the DNA sequence such as the chromatin organization.
Chromatin is a macromolecule that helps packaging DNA and proteins to make them fit within the cell.
It also serves as an index system to organise the genome.
Figure~\ref{pic_epigenetics} depicts the concept of chromatin.
There are two chromatin states, open and closed chromatin, which facilitate the DNA to be more or less compact.
Hence, the states effect the gene expression, as for genomic regions with close chromatin the DNA is less accessible to transcription factors (TFs) compared to regions with open chromatin. 
Chromatin states play a key role for example in cell differentiation by allowing the selective expression of particular genes.

A nucleosome is the fundamental core unit of chromatin and consists of eight histones.
Histones are proteins where the DNA is wrapped around to be spatially organised in a maximal condensed way.
There are two types of histones: core histones form the nucleosome and linker histones bind the nucleosome to the DNA.
Amino-terminal histone tails drive through the nucleosome core and make contact with adjacent nucleosomes to build the chromatin structure.

Enzymes may chemically modify the histone tails which then affect the overall chromatin structure.
The enzyme adds a chemical flag to the histone tails comparable to DNA methylation or the chromatin remodelling processes.
We refer to a histone with a particular chemical flag as histone modification.
Importantly, some histone modifications effect the chromatin which effects the gene expression.
Particular histone modifications are therefore associated for example to activation or deactivation of gene expression (see Figure~\ref{pic_epigenetics}).

The naming of a histone modifications follows the following structure: the histone number in a nucleosome whose tail is modified, the single-letter amino acid abbreviation, the amino acid position in the protein, the type of the modification and the number of modifications.
For instance, the histone modification H3K4me3 describes a chemical modification of histone three (H3), where the amino acid lysine at the fourth position (K4) is changed by adding three methyl groups (me3).
Histone modifications H3K79me2 and H3K36me3 are associated to transcription~\citep{Nguyen2011, Sims2099}.
Histone modification H3K4me3, typically located in the promoter, H3K4me1, typically located in enhancer regions, as well as H3K27ac and H3K9ac are associated to activation~\citep{Briggs2001, Creyghton2010, Grant1999}.
Histone modification H3K9me3 and H3K27me3 are associated to repression~\citep{Cao2002}.

\begin{figure}[ht]
  \centering
  \includegraphics[width = 10cm]{pics/epigenetics}
\caption[Epigenetic concept]{Epigenetic concept. 
Among others, gene expression is regulated by closed and open chromatin.
Closed chromatin exhibits repressive histone modifications and inhibits RNA ploymerases II to attach to the DNA.
In contrary, open chromatin has active histone marks as well as certain activator proteins.
Activator proteins are gene regulatory proteins which are also shown in Figure~\ref{pic_gene_transciption}. 
They interact with mediator proteins and enable transcription factors as well as RNA polymerase to bind to the DNA.
RNA polymerase reads the DNA from 5' to 3' end.
Thereby, RNA molecules are produced which eventually are translated to proteins.
Histone modifications are also associated to DNA transcription.
The figure is based on \cite{Lodish2007}.
}
\label{pic_epigenetics}
\end{figure}




\section{ChIP-seq to Analyze Epigenetics}
We introduce chromatin immunoprecipitation followed by sequencing (ChIP-seq), a method to identify DNA-protein complexes in a genome wide manner.
First, we explain ChIP which is a method to isolate DNA fragments that are attached to certain proteins of interest.
We then explain DNA sequencing which is used to determine the nucleotide sequence of a DNA molecule and its position with regard to a reference genome.
Finally, we describe ChIP-seq which combines ChIP with DNA sequencing.
Further, we emphasize important challenges that have to be considered in the analysis of ChIP-seq experiments.

\subsection{ChIP}
\label{sec_chip}
Chromatin Immunoprecipitation (ChIP) is used to investigate protein-DNA interactions inside a cell.
In particular, ChIP enables localization of posttranslational modifications of the histone tails (see Section~\ref{sec_epigenetics}) as well as DNA target sites of TFs in the genome.
The basic idea of ChIP was first reported in the 1960s~\citep{Collas2010}, while applications of ChIP in studies of histone-DNA interactions go back to the late 1970s~\citep{Jackson1978}.

The ChIP protocol has several steps.
First, DNA and proteins in a cell are cross-linked with formaldehyde.
These DNA-protein complexes are fragmented by for example sonication into fragments of $200-1000$bp.
Specific antibodies are then used to pull out (immunoprecipitate) protein-DNA complexes that contain the proteins of interest. 
Finally, the cross-link is reversed (formaldehyde is heat-reversible) and the DNA that was bound to the protein is purified.
To identify the DNA sequences associated with the proteins of interest, downstream analysis, such as DNA sequencing (see Figure~\ref{pic_chip_seq_workflow}), is required.
% The DNA microarray is one of the most commonly used methods~\citep{Collas2010}.
% Several known DNA sequences (targets) are attached to a solid surface.
% To determine the DNA sequence of interest, hybridization with target DNA is performed and quantified.
% The combination of ChIP and microarray analysis is called ChIP-on-chip method~\citep{Lee2006}.

\subsection{DNA sequencing}
\label{sec_dna_sequencing}
DNA sequencing methods determine the base sequence of a DNA sample.
The first approach to sequence DNA was the chain-termination method invented by \cite{sanger:1977}.
% This method is able to create DNA sequences (reads) with a length up to 1000bp, but offers some crucial disadvantages: Sanger-sequencing needs a lot of space, time and money to produce results.
However, routine studies of mammalian became possible by high-throughput sequencing technologies which are also called next-generation sequencing (NGS) or second-generation sequencing.
NGS takes advantage of parallelization: nucleotides are read in parallel and a high number of DNA fragments is considered at the same time~\citep{shendure:2008}.
Parallelization is often done by cloning DNA fragments which is usually performed by PCR~\citep{Mullis1987}.
Each DNA fragment can be sequenced from one end, resulting in single-end reads, or from both ends, resulting in paired-end reads.
In most cases, the reads are shorter than the DNA fragments and therefore give only partially the base sequence of the fragment.
Compared to Sanger sequencing, the costs of NGS are much lower, while the base calls are less accurate and the reads are smaller.
NGS is the current method for large scale sequencing applications.
In many cases, a reference genome of the organism is known.
Sequencing of organisms with a given reference genome results in the computation problem to determine the position of each read in the genome.
We refer to the process of estimating the read positions as aligning or mapping the reads to the reference genome.
There are various technologies for DNA sequencing.
All sequencing experiments analyses in this thesis were performed with Illumina devices.


\subsection{ChIP-seq Method}
\label{sec_chip_seq}
ChIP-seq combines the ChIP protocol with high-throughput sequencing technologies and thereby offers a low-cost way to identify DNA-protein interactions in a genome wide manner~\citep{park2009}. 
ChIP-seq was one of the early applications of NGS~\citep{Johnson2007}.
Figure~\ref{pic_chip_seq_workflow} gives an overview of the ChIP-seq workflow.
First, the DNA obtained by ChIP (see Figure~\ref{pic_chip_seq_workflow}, step 1) and associated to the proteins of interest is sequenced by NGS methods (see Figure~\ref{pic_chip_seq_workflow}, step 2).
NGS sequencing typically produces short (\texttildelow $50-100$bp) single-ended reads.
The reads are then aligned to the reference genome~\citep{park2009} (see Figure~\ref{pic_chip_seq_workflow}, step 3).
Finally, a genomic signal is created based on the aligned reads (see Figure~\ref{pic_chip_seq_workflow}, step 4). 
Genomic regions where reads accumulate more than by chance (peaks) are identified within the ChIP-seq signal (peak calling).
Peaks represent regions in the genome where the proteins of interest are localized in the original DNA sample~\citep{Collas2010}.

The size of DNA fragments is larger than the protein-DNA interaction site. 
Therefore, reads derived from the DNA fragments map to different genomic locations which results in fuzzy peak shapes.
Furthermore, peaks usually have different shapes due to the underlying proteins of interest taken into account by the ChIP protocol.
TFs usually attach to small DNA regions without any further TF in the vicinity.
The ChIP-seq landscape of TFs therefore tends to exhibit sharp, isolated peaks.
In contrary, histone modifications are organised in groups where all histones are close to each other.
In general, this leads to a complex ChIP-seq landscape with several peaks in close vicinity.
Histone modifications at active regulatory elements exhibit relatively small protein domains.
Broad domains are caused by a high number of proteins, which typically occur for histone modifications that repress genomic regions.
% Mixed peaks are associated with proteins which on the one hand bind to specific motifs in the DNA sequence but on the other hand can also be detected over the body of actively transcribed genes~\citep{park2009}.

\begin{figure}[ht]
  \centering
  \includegraphics[width = 15cm]{pics/chipseq_workflow.pdf}
\caption[ChIP-seq workflow]{ChIP-seq workflow. 
We show DNA that is wrapped around histones which may or may not be modified.
The histone modifications are indicated by a green signal at their tails.
First, ChIP is used to fetch out the proteins of interest with specific antibodies and to shear the DNA.
Antibodies are represented as elements that are attached to the modified histones.
Next, NGS methods are used to create reads which are mapped to a given reference genome.
As the DNA fragment is not entirely sequenced, the reads stem either from the beginning or the end of the fragment.
Reads can be mapped to the forward strand, that is, left oriented reads are mapped from the 5' to 3' end of the genome; or to the reverse strand, that is, right oriented reads are mapped from 3' to 5' end of the genome.
Typically, a discrete signal is then derived from the set of aligned reads for the entire genome.
It is a common procedure to extend the ChIP-seq reads to match the fragment size.
Peaks indicated by red boxes in the ChIP-seq signal refer to positions in the genome where DNA interacts with the proteins of interest.
}
\label{pic_chip_seq_workflow}
\end{figure}

\subsection{Control Sample}
\label{sec_control_sample}
Each experimental step in the ChIP-seq protocol potentially involves various sources of artifacts.
For example, DNA shearing usually does not result in a uniform fragment distribution of the genome, as open chromatin regions tend to be fragmented more easily than closed regions.
Also, repetitive regions in the sample DNA may incorrectly seem to be enriched due to differences in the reference and sample genome~\citep{park2009}.
Moreover, the DNA to be analysed may be contaminated with DNA that was not bound by the chosen antibody.
Therefore, it is highly recommended to compare a peak in a ChIP-seq profile to a control sample to determine its significance in the same cell~\citep{Meyer2014}.
There are three different ways to obtain control DNA:
\begin{itemize}
 \item input-DNA: a fraction of the DNA sample is removed prior to the immunoprecipitation step;
 \item mock IP DNA: DNA is obtained from immunoprecipitation without an antibody; and
 \item DNA from non-specific immunoprecipitation, that is, the immunoprecipitation step is performed for a fraction of the sample DNA with an antibody that is known to not be involved in DNA bindings.
\end{itemize}
Input-DNA is the most widely used method as it is assumed to test against the most common artifacts introduced by the ChIP-seq protocol such as bias in the DNA fragmentation process ~\citep{park2009, Furey2012}.




\subsection{Arising Challenges}
\label{sec_chipseq_challenges}
The ChIP-seq protocol is frequently refined to improve its accuracy~\citep{Meyer2014}.
Various challenges arising from the technical and computational side have to be overcome to improve the peak detection.
Here, we list the most important challenges associated to ChIP-seq.

% is this fitting here?
% These applications require the joint analysis of ChIP-seq data with or without replicates. 
% However, ChIP-seq is a multi-step experimental protocol and each of this steps introduce distinct sources of potential artifacts~\cite{Meyer2014}. 
% Among others, these artifacts arise from bias of DNA fragmentation to open chromatin regions, variation of IP efficiency due to antibodies, as well as PCR amplification and sequencing depth bias. 
% These artifacts produce ChIP-seq experiments with distinct signal-to-noise ratios, even when they are produced in the same lab and follow the same protocols~\cite{Furey2012,Meyer2014}. 
% Moreover, the analysis of clinical samples, where patients have a distinct genetic background and samples may arise from heterogeneous cell populations, introduces further sample specific variation to the ChIP-seq signals~\cite{ashoor2013}.
% All these artifacts impose great challenges to the computational analysis of ChIP-seq data.

\subsubsection{Antibody}
\label{sec_challenges_antibody}
The antibody's sensitivity and specificity chosen for the ChIP experiment is crucial for the analysis, as the antibody directly affects how well defined a peak in the ChIP-seq signal appears. 
High quality antibodies precisely pull out the proper protein-DNA complexes and thereby ensure a high level of enriched signal compared to the background noise. 
However, antibody quality can vary even between the same biological experiment using the same antibody~\citep{park2009, Furey2012}.
Also, for some proteins there is no proper antibody, such that these proteins cannot be examined by ChIP-seq.
Furthermore, the quality of the antibody may also depend on the manufacturer. 

\subsubsection{Cell Population}
\label{sec_challenges_cell_population}
A typical ChIP experiment needs approximately $10^7$ cells and thereby limits the number of ChIP experiments that can be performed on a biological sample.
The number of cells depends on the quality of the antibody as well as the abundance of the target protein~\citep{Furey2012}.
Some techniques~\citep{Acevedo2007, Adli2011} have been developed to decrease the number of required cells.
However, fewer cells generally produce less well defined peaks in the resulting ChIP-seq signal.
% Also, single molecule sequencing techniques may help to reduce the required cell number~\citep{Harris2008, park2009}.

\subsubsection{Sequencing}
\label{sec_challenges_sequencing}
DNA fragments obtained from ChIP are sequenced and mapped to a reference genome. 
The sequencing depth is crucial for the success of the ChIP-seq experiment. 
It is recommended to have approximately $2 \cdot 10^7$ reads for ChIP-seq experiments with the human genome and target proteins that lead to relatively isolated ChIP-seq peaks such as active histone marks. 
Apart the absolute number of reads, it is recommended that the amount of reads mapping to distinct genomic location is higher than $80\%$~\citep{Furey2012}.

There are genomic regions that cannot be captured in the ChIP-seq analysis~\citep{Dunham2012}.
These regions comprise unstructured, high signals and occur independently of the type of NGS experiment.
We refer to such regions as blacklisted regions.
They are typically ignored in the ChIP-seq analysis.

Several studies address sequencing specific bias~\citep{Khrameeva2012, Allhoff2013} which have to be taken into account by the mapping algorithm and the downstream analysis.
Particularly important is the bias due to \nuc{GC}-content.
\cite{Benjamini2012} describe a dependency between the fragment count and the fragment's \nuc{GC}-content which may lead to artificial, non-biological high signals in the ChIP-seq experiment.

\subsubsection{PCR Duplicates}
\label{sec_challenges_pcr_duplicates}
PCR duplicates lead to artificially induced reads that impose unwanted bias in the downstream analysis.
There are two ways how these duplicates are created.
First, during PCR (see Section~\ref{sec_dna_sequencing}) of the sequencing procedure, duplicates may be created by accidentally considering several times the same fragment.
Second, PCR duplicates may be created in the picture analyzing step during the sequencing process.
That is, one DNA feature is mistaken as two or more features~\citep{Meyer2014, Maze2014}.
Both types of PCR duplicates lead to reads which are mapped to the same genomic location.
However, because of sonication based fragmentation, it is highly unlikely that two DNA fragments will stem from the same genomic location.
Hence, reads with identical mapping positions indicate that they are PCR duplicates.

\subsubsection{Fragment Size Estimation}
\label{sec_challenges_shift}
ChIP-seq experiments typically comprise single-ended reads.
For all DNA fragments, these single-ended reads are expected to come from on average uniform ratio of both DNA strands.
Furthermore, the reads partially cover only one end of the fragments.
% Section~\ref{sec_chip_seq} describes that either the start or the end of the DNA fragment is sequenced.
Hence, the read distribution exhibits two peaks up- and downstream of the proteins of interest (see Figure~\ref{pic_fragment_size_estimate}).
Typically, the reads are extended to the original fragment length, such that the read distribution provides a single peak that correlates with the protein position.

The fragment length can be obtained from the ChIP-seq protocol or be computationally estimated with the aligned ChIP-seq reads.
Due to some expected difficulties in the protocol's accuracy, the fragment size is usually derived from the reads~\citep{Pepke2009}.
In Figure~\ref{pic_chip_seq_workflow} and Figure~\ref{pic_fragment_size_estimate}, the extension size of reads is indicated by a distance arrow for each read.
The extended reads are then used to generate the genomic signal (Step 4 in Figure ~\ref{pic_chip_seq_workflow}).

\begin{figure}[ht]
  \centering
  \includegraphics[width = 9cm]{pics/fragment_size_estimate.pdf}
\caption[Fragment size estimation]{Fragment size estimation. 
Reads are mapped to the forward strand (black line) or reverse strand (dotted black line) of the reference genome.
The distributions of the reads (red and pink) build two peaks at the left and right side of the protein of interest, as only the beginning or the end of the DNA fragments is sequenced.
The original DNA fragments are indicated as dotted lines which extend the reads.
Because of the shearing process of the ChIP-seq protocol, the fragments slightly differ in their start positions.
% However, all fragments cover the protein of interest.
The fragmentation size is computed and the reads are artificially extended to obtain the original fragment length.
The distribution of the extended reads (red) exhibits one peak whose position correlates with the position of the protein of interest.
The figure is based on~\cite{park2009}.
}
\label{pic_fragment_size_estimate}
\end{figure}



\subsection{Quality Measures of ChIP-seq Experiments}
\label{sec_eval_chipseq_exp}
Successfully calling peaks in a ChIP-seq signal highly depends on the signal's signal-to-noise ratio~\citep{landt2012}.
The ChIP-seq signal correlates to DNA-protein interaction sites.
Signal that is not correlated to the these interaction sites is called background noise.
Background noise may stem from various sources such as the fragmentation step or poor antibodies in the ChIP protocol.
A high signal-to-noise ratio is highly desired for all downstream analyses as it positively effects the accuracy of the peak calling step.

\cite{landt2012} introduced several widely used measures to evaluate ChIP-seq data. 
These metrics give indications about the quality of the ChIP-seq experiment. 
First, the fraction of reads in peaks (FRiP) is an indicator for the signal-to-noise ratio in the data.
The FRiP is estimated by calling peaks in a ChIP-seq signal and computing the ratio of reads within the called peaks and the overall number of reads.
The higher the FRiP, the better the signal-to-noise ratio.
See Figure~\ref{pic_ratio_to_noise} for an illustration of FRiP.
% Second, the absolute number of unique reads has to be sufficiently high to ensure covering all genomic regions with the proteins of interest (see also Section~\ref{sec_challenges_cell_population}).
Second, the non-redundant fraction (NRF) of reads is the ratio between the number of positions in the genome that uniquely mappable reads map to and the total number of uniquely mappable reads. 
NRF is associated with the number of PCR duplicates (see Section~\ref{sec_challenges_pcr_duplicates}) and measures the entropy of the set of aligned reads.
NRF decreases with sequencing depth as at some point PCR-amplified DNA fragment will be repeatedly sequenced.

\begin{figure}[ht]
  \centering
%   \includegraphics[width = 14cm]{pics/example_ratio_to_noise.pdf}
    \includegraphics[width = 14cm]{pics/example_ratio_to_noise.png}
\caption[Example for different signal-to-noise ratios]{Example for different signal-to-noise ratios.
The figure shows two ChIP-seq profiles of histone modification H3K27ac based on two cancer patients of the same cell.
We therefore assume that the ChIP-seq profiles have similar peaks.
ChIP-seq profile S1 has a low and profile S2 a high signal-to-noise ratio (please note the different y-axis scales).
To compute FRiP, peaks (black bars below the signal) are first called for S1 and S2.
Next, the number of reads falling into these peaks is divided by the total number of reads.
% As S1 and S2 are based on the same cell, we additionally show the consensus peaks (black bars) between both signals.
The ChIP-seq data stem from \cite{Koues2015}.
}
\label{pic_ratio_to_noise}
\end{figure}


\section{Computational Analysis of ChIP-seq}
Computational analysis is necessary to derive the positions of DNA-protein complexes from ChIP-seq data.
First, we introduce the single peak calling problem.
Next, we formalize the differential peak calling problem and motivate it by presenting related studies of current biological and medical research.
Finally, we point out arising challenges.

\subsection{Single Peak Calling Problem}
\label{sec_spcp}
A common goal in ChIP-seq data analysis is the genome wide detection of protein-DNA interactions in a single biological condition.
The protein-DNA interaction positions are associated to peaks in a ChIP-seq profile.
We refer to the detection of peaks in a ChIP-seq profile as the single signal peak calling problem.

\begin{mydef}[Single Signal Peak Calling Problem]
  For a given ChIP-seq profile $S$, find genomic position (peaks) where the signal is significantly enriched.
\end{mydef}

\noindent
We assign a discrete value to each genomic location, where the value corresponds to the strength of the protein binding event.

Single Peak Callers (SPCs) typically work in two phases.
First, they segment the genomic signal into background regions and regions with potential peaks.
The segmentation is either performed with a window-based approach or more sophisticated methods like Hidden Markov Models (HMM).
Then, they perform a statistical test to check whether the potential peaks significantly differ to the background signal.
SPC provide a list of peaks where each peak is typically assigned to a $p$-value.

SPCP has already been addressed by several research groups.
\cite{wilbanks2010} as well as \cite{Chen2012} review and evaluate various SPCs.
SPCs with good evaluation performance in transcription factor binding sites (TFBS) studies are for example PeakSeq~\citep{Rozowsky2009}, QuEST~\citep{valouev2008} and MACS~\citep{zhang2008}.
Moreover, sophisticated segmentation methods like hidden Markov models (HMM) are used for example by HMCan \citep{ashoor2013} and BayesPeaks~\citep{spyrou2009}.
Figure~\ref{pic_dp_example} gives an example for the peak prediction of a SPC.
Light blue stripes below the ChIP-seq profiles indicate where the SPC calls a peaks.

ChIP-seq experiments are often replicated to avoid considering peaks resulting from variability by random chance.
Replication is therefore desired to distinguish between biological and random events as well as to verify the reliability of experimental steps~\citep{park2009}.
The majority of SPCs is not able to handle replicates and only recently, strategies have been developed for that purpose.
For example, the ENCODE project proposes the use of the irreproducible discovery rate (IDR).
IDR finds common peaks of a set of candidate peaks that are separately called by SPCs on individual replicates~\citep{landt2012,Li2011}. 
Also,~\cite{Ibrahim2015} propose a method for the joint analysis of ChIP-seq replicates for SPCP. 
Their method detects peak boundaries with higher precision than identifying common peaks in replicates with IDR or pooling ChIP-seq reads of replicates.

\subsection{Differential Peak Calling Problem}
Differential peak calling is an important challenge in current medical and biological research that investigates changes in protein-DNA interactions of distinct cellular conditions.
In contrary to SPCP this computationally challenge has not been extensively addressed.
% Moreover, replicates play an important role.
The differential peak calling problem is defined as following:

\begin{mydef}[Differential Peak Calling Problem]
  Given two experimental conditions $S_1 = \{S_{11}, \ldots, S_{1k}\}$ and $S_2 = \{S_{21}, \ldots, S_{2k}\}$ containing a set of genomic ChIP-seq signals, find genomic positions (differential peaks) where $S_1$ and $S_2$ significantly differ.
\end{mydef}

\noindent
We are interested in significant differential peaks (DPs) between two biological conditions $S_1$ and $S_2$ which can or cannot contain replicates.

\begin{figure}[ht]
  \centering
  \includegraphics[width = 12cm]{pics/dp_example_without_rep}
\caption[Differential peak calling example]{Differential peak calling example. 
We show an example of two distinct ChIP-seq signals for the histone modification H3K4me2 before (0h, upper signal) and 24 hours after (24h, lower signal) induction of TLR4 signaling of macrophages around the gene Irf1~\citep{kaikonnen2013}. 
We indicate with squares examples of regions, which are putative DPs with gain (or loss) of ChIP-seq signal after 24 hours of TLR4 treatment. 
The height of the squares indicates the size of the highest ChIP-seq signal for a DP. 
We display results from the SPC PeakSeq (grey bars) and a two-stage peak caller based on applying DESeq on PeakSeq peaks (black bars). 
PeakSeq successfully detects broad peaks describing ChIP-seq signal for each cell. 
The two-stage peak caller can detect DP1 and DP3, but cannot detect changes within the broad candidate peaks such as DP2 or complex changes in the signal within the Irf1 gene body (DP4 and DP5). 
}
\label{pic_dp_example}
\end{figure}

Initially, differential peak calling was performed by peak calling on individual ChIP-seq signals. 
Peaks detected in only one of the conditions were then defined as cell-specific peaks~\citep{Heinz2010}. 
Such methods are not able to detect cases where peaks were presented (and called) in both cell types, but exhibit a significant increase (decrease) of the DNA-protein signal in one of the cells. 
In the example of Figure~\ref{pic_dp_example}, which is based on peaks from PeakSeq~\citep{Rozowsky2009}, only DP1 would be detected as cell-specific.  
Moreover, most SPCs provide no functionality to normalize several ChIP-seq experiments and are likely to show bias in experiments with distinct number of reads. 

A more sophisticated strategy to detect DPs is the combination of peaks from SPCs with statistical methods for the analysis of differential gene expression of RNA-seq data.
These two-stage differential peak callers (DPCs) first combine peaks that are called on individual ChIP-seq conditions using SPCs.
Next, they count the number of reads for each candidate peak, perform signal normalization and apply statistical tests assuming a differential count model. 
Therefore, they can detect candidate peaks where the number of read counts is significantly higher or lower in one of the ChIP-seq conditions. 
While this approach allows the detection of significant changes in ChIP-seq within candidate peaks, it is highly dependent on the initial peak calling step as well as the strategy used to create the set of candidate peaks. 
For example, histone modifications associated to active regulatory regions occur in domains spanning several hundreds of base pairs and may have intricate patterns of gain/loss of ChIP-seq signals within the same domain. 
SPCs tend to call the domains as single peaks and consequently the differential analysis is only able to evaluate the differential counts of the complete called peaks.
In Figure~\ref{pic_dp_example}, the SPCs calls one peak for the gene body of Irf4.
The DPC is therefore not able to distinguish between DP4 and DP5.
% A more  appropriate framework to analyse DPs is the use of segmentation methods like Hidden Markov Models (HMMs), which are able to analyse pairs of ChIP-seq signals and perform DPC in a single step~\citep{xu2008}. 
% There is a clear need for methods that accurately solve DPCP.

A further strategy to detect DPs is to first segment the genome with a fixed window.
Next, it is tested whether the windows contain differential counts.
Heuristics methods are applied to merge windows in close vicinity to each other and with a similar counts~\citep{Li2013}.
The performance of such methods depends on the window merging strategy as well as the window size.
Too large windows tend to omit small peaks in the ChIP-seq signal.
In Figure~\ref{pic_dp_example}, DP2 is not detectable with a window size of 1000bp, which is the default parameter of the DPC Diffreps~\cite{Li2013}.


\subsection{Example of Studies Comparing ChIP-seq Signals}
There are several studies of current biological and medical research that compare ChIP-seq signals under distinct conditions.
These studies investigate for example

\begin{itemize}
 \item cell differentiation: \cite{Lin2015} investigates regulatory changes in a mouse model during the development of antigen-presenting dendritic cells with regard to the histone modification H3K4me1 and H3K27ac.
 \item cell activation: \cite{Stunnenberg2014} perform ChIP-seq experiments in humans for monocytes that are activated to ma\-cro\-pha\-ges.
 The differentiation from monocytes to ma\-cro\-pha\-ges plays a key role in the host's defence system. 
 The study describes epigenetic differences with regard to several histone modifications.
 Biological replicates based on different donors are used for the study.
 \item comparison of healthy and diseased individuals: \cite{Koues2015} analyse the difference of regulatory genomic features between healthy individuals and lymphomas patients. 
 They investigate the histone modification H3K27ac.
 \item the activation of signaling pathways: \cite{kaikonnen2013} describe the response of macrophages after the time dependent activation of the TLR4 pathway which plays an important role in the immune system.
 Their study comprises a mouse model and does not provide replicates.
\end{itemize}

\noindent
Calling DPs in these studies can give findings that lead in general to a deeper understanding of epigenetics.
Depending on the application, DP predictions may exhibit starting points for drug discovery and epigenetic biomarker detection~\citep{Koues2015, Stunnenberg2014}, give new findings of cell differentiation steps~\citep{Lin2015} or unravel mechanisms for the immune system activation~\citep{kaikonnen2013}.

% These methods should be robust against different data characteristics like for example the variance within replicates.


\subsection{Arising Challenges}
\label{sec_challenges_dpcp}
The differential peak calling problem leads to computational challenges which arise additionally to the ChIP-seq specific tasks described in Section~\ref{sec_chipseq_challenges}.

\subsubsection{Replicates}
\label{sec_challenges_rep}
Replicated ChIP-seq experiments can be used to reduce the effect of unwanted technical bias.
There are two kinds of replicates of ChIP-seq experiments.
Biological replicates stem from independent cell cultures or tissue samples to ensure reproducibility.
Technical replicates are based on measuring a single biological sample and can therefore only be used to estimate the variability of the sequencing step~\citep{Yang2014}.
% It has been shown that two biological replicates are usually sufficient as a third replicates does not lead to further information about the sample variability~\citep{Rozowsky2009}.
% % cell population: This is in particular important for using ChIP-seq replicates (see Section~\ref{sec_replicates}).
These two kinds exhibit different characteristics.
For instance, the variance between biological replicates is supposed to be higher than for technical replicates, as they stem from various biological samples.

If replicates are available, the problem becomes computationally more complex as in general more information has to be taken into account.
% , different signal-to-noise ratios of replicates have to be normalized to ensure accurate peak estimations.
% The higher this condition specific variability is, the harder the DPs can be found. 
In particular, count data derived from NGS data usually exhibits overdispersion, that is, the variance in the data exceeds the mean~\citep{anders2010, Cameron1999, Ismail2007}.
To ensure accurate DP estimates, overdispersion has to be taken into account which requires an appropriate complex statistical model.
% The general effect of overdispersion in count data is already well described~\citep{Cameron1999, }.
% To model count data appropriate distributions such as the Negative Binomial distribution have to be chosen.

\subsubsection{Normalization}
\label{sec_challenges_norm}
For the differential peak calling problem, we compare several ChIP-seq profiles which typically exhibit different sequencing depths as well as different signal-to-noise ratios.
Normalization against different sequencing depths is necessary, as otherwise ChIP-seq profiles may be over- or underrepresented when comparing them.
The signal-to-noise ratios should also be considered in the normalization.
Even in the case without replicates, normalization of samples associated to distinct conditions is important.

\subsubsection{Evaluation}
\label{sec_challenges_eval}
There is no direct metric to systematically quantify DP predictions.
Furthermore, due to the biological complexity, there is no genome wide map of DNA-protein interactions which could be used as a gold standard.
Consequently, evaluating solutions for the differential peak calling problem is still an open problem.
However, indirect metrics can be used to quantify the DP predictions.
For example, as gene expression correlates well to certain histone modifications~\citep{Karlic2010}, the validation of DP predictions with gene expression is possible.
Furthermore, the simulation of ChIP-seq reads is an effective strategy to produce artificial gold standards with various data characteristics~\citep{humburg2011, zhang2008, Aaron2014}.


\section{Related Work}
The differential peak calling problem has only been addressed in a few studies. 
Here, we first review normalization approaches.
Second, we give an overview of existing simulation algorithms to evaluate DPCs.
Finally, we list existing methods to solve the differential peak calling problem and give a short description of their working procedure.
We explain how the tools deal with the challenges described in Section~\ref{sec_challenges_eval} and Section~\ref{sec_chipseq_challenges}.

\subsection{Normalization}
\label{sec_back_norm}
The majority of normalization approaches multiplies the ChIP-seq signals by a factor.
One example is the normalization by library sizes. 
Here, the normalization factor is the ratio between the total number of counts of the signal with the highest total number of counts; and the total number of counts of the signal which has to be normalized.
Thereby, signals with lower total counts are raised to the level of the signal with maximal total counts.

We demonstrate the normalization with an example.
We resort to Figure~\ref{pic_ratio_to_noise} which shows two replicates of the same condition, but with different read counts and signal-to-noise ratios.
As the replicates stem from the same condition, the consensus peaks should have similar counts across the conditions after normalization.
MA plots visualize count distributions in two genomic signals and we use them to picture the counts of the consensus peaks in Figure~\ref{pic_ratio_to_noise}.
We divide the genome into consecutive bins and count the signal specific reads falling into these bins, that is, for each bin, we obtain the number of reads for both signals.
The MA-plot assigned for each bin the M-value, that is, the logarithmic ratio of the counts, to the A-value, that is, the logarithmic mean of the counts.
Figure~\ref{pic_norms_maplot}A gives the MA-plot for the signals shown in Figure~\ref{pic_ratio_to_noise}.
The rationale for using MA-plots is that, after signal normalization, the bins associated to consensus peaks (indicated by red points) should give low M-values, as they stem from two replicates of the same condition.
Without any normalization, the mean M-value of peak associated bins (MMP) is $2.4$.

In this example, the normalization by library sizes gives a factor of $1.6$ for signal S1.
S1's low signal-to-noise ratio inhibits a higher normalization factor, as the entire signal of S1, including the noise, is used for the calculation.
Figure~\ref{pic_norms_maplot}B gives the corresponding MA-plot.
Compared to the case without normalization, the peak associated bins yield a lower MMP value ($1.54$), which demonstrates the advantage of the normalization.

\cite{Robinson2010norm} propose a strategy to normalize RNA gene expression data under the assumption that the majority of the genes are not differentially expressed. 
For given signals, they first compute the M- and A-values.
Next, they estimate a quantile based range of the values which are used for the normalization.
The rationale of not considering outliers of M- and A-values is that they may have a strong influence to the results~\citep{Meyer2014}. 
The normalization factor is computed by the product of the resulting M- and A-values normalized against the A-values.
They refer to their normalization approach as trimmed mean of M-values (TMM).
\cite{anders2010} implemented an similar approach by using the geometric mean of the gene expression data.
The majority of DPCs dealing with replicates use a TMM-based normalization strategy. 

Figure~\ref{pic_norms_maplot}C depicts the MA-plot after normalizing with a TMM-based factor of $1.29$ for S1 and $0.98$ for S2.
Trimming M- and A-values does not exclude the noise signal which is still comprehensively considered for the estimation of the normalization factors.
Hence, the TMM normalization leads to an MMP of $1.79$ which is in this example even higher than the MMP of the simple normalization by library sizes ($1.54$).

% However, TMM was devised for gene expression experiments which assumes that counts of most observations (genes or peaks) do not change.
% This is not necessarily the case for protein interactions, as two distinct cells can have distinct amounts of proteins or histone modifications bound to their DNA~\citep{Meyer2014}. 
% Particularly problematic in this normalization approaches is the effect of replicate specific background noise. 

\begin{figure}[ht]
  \centering
%   \includegraphics[width = \textwidth]{pics/norm_MA_plots.pdf}
  \includegraphics[width = \textwidth]{pics/norm_MA_plots.png}
\caption[MA-plots for different normalization approaches]{We show MA-plots for different normalization approaches for the region shown in Figure~\ref{pic_ratio_to_noise}, where we apply IDR to find common peaks within the replicates.
% The replicates are chosen because of their high difference in FRiP.
Bins associated to IDR-peaks are highlighted in red.
We also give the mean M-value of peak associated bins (MMP) which is supposed to be low after normalization. 
We depict MA-plots without (A), after the normalization by library sizes (B) and after the TMM (C) normalization.
}
\label{pic_norms_maplot}
\end{figure}

\subsection{Evaluation}
\label{sec_sim_previous_work}
There is neither a direct metric to rate nor a gold standard to compare DP predictions.
However, the simulation of ChIP-seq profiles is an effective strategy to evaluate DPCs.
The majority of ChIP-seq simulators are based on SPCP.
\cite{zhang2008} developed a strategy to model TF-based ChIP-seq signals that contain sharp peaks. 
They simulate the background noise with a Gamma distribution which determines the impact of noise on particular genomic regions.
\cite{zhang2008} do not provide NGS reads and consequently they lack to model the bias based on the sequencing process, such as the \nuc{GC}-content.
\cite{humburg2011} followed \cite{zhang2008} and extended their model to make it capable of producing NGS reads.
He enhanced the model by making it more flexible in terms of the number of reads and the number of binding events.
However, \cite{humburg2011} does not specifically model the number of binding events.
None of these approaches can be used directly for the differential peak calling problem as they are restricted to exactly one ChIP-seq profile.

\cite{Aaron2014} developed a method to simulate ChIP-seq profiles with DPs between two conditions.
The reads of an enriched regions were sampled from a Negative Binomial distribution.
DPs are included by adjusting the parameters of the Negative Binomial distribution such that one condition is expected to gain more reads than the other.
Next, the reads' positions are determined.
The simulation algorithm lacks to model crucial parameters such as the background noise and the variability of peaks in ChIP-seq profiles associated to the same condition.
Also, their simulation algorithm is not public available. 

\subsection{Two-Stage Differential Peak Caller}
\label{sec_previous_twostage_dpcs}
DPCs can be roughly categorized in two-stage and one-stage DPCs. 
Two-stage approaches are based on separate candidate peaks for each ChIP-seq profile.
These candidate peaks are pre-computed by SPCs and are used as input for sophisticated differential count models.
In general, two-stage DPCs merge the candidate peaks with regard to the conditions they stem from, count the number of reads for each candidate peak, perform signal normalization and apply statistical tests assuming particular count models.
Some two-stage DPCs use count models that are tailored for the differential expression analysis of RNA-seq data.
Such models are for instance implemented in DESeq~\citep{anders2010} and edgeR~\citep{robinson2010}.
We list all two-stage DPCs that are available to our knowledge.
Table~\ref{tab_tools} gives an overview of the tools and their supported features.

\subsubsection{DiffBind}
DiffBind~\citep{Start2013} is a two-stage differential peak method based on SPC candidate peaks.
First, the peak lists are merged to obtain consensus peaks.
The number of reads falling in to these consensus peaks are counted and a statistical model based on edgeR~\citep{robinson2010} is estimated to call DPs.
% The tool edgeR was designed to identify significant differences in gene expression data. 
The count data is modelled by a Negative Binomial distribution to take overdispersion induced by potential replicates into account.
DiffBind normalizes data by following the TMM approach after input-control is subtracted from ChIP-seq profiles.
DiffBind can take replicates into account.
However, neither the fragmentation size nor \nuc{GC}-content is estimated by DiffBind.
Also, the input-DNA is not normalized and no postprocessing step is implemented to filter artefacts.

\subsubsection{MACS2}
MACS2~(unpublished, available at \url{https://github.com/taoliu/MACS/}, last access October 14th, 2015) works in two steps.
First, all ChIP-seq profiles are pooled together and MACS2's SPC (\textit{callpeak}) is executed for each condition.
Second, we use MACS2's algorithm \textit{bdgdiff} to identify DPs within these peaks.
The SPC normalize against input-DNA, considers \nuc{GC}-content and estimates the fragmentation size.
MACS2's differential peak calling method works by a sliding window approach on candidate regions (personal communication). 
There is no formal description of its parameters and the strategy for normalization. 

\subsubsection{DESeq}
We combine DEseq with SPCs to make it applicable to the differential peak calling problem.
First, a SPC computes a list of candidate peaks for each condition and second, DESeq is used to determine DPs.
DESeq~\citep{anders2010} is a tool to analysis differential gene expression.
The observed counts are normalized with the geometric mean and the count data is modelled with a Negative Binomial distribution.
DESeq uses the Negative Binomial distribution to compute a $p$-value for each estimated differential gene.
By using the Negative Binomial distribution, DESeq is capable to take overdispersion into account.
In general, combinations of DESeq and a SPC does not apply any filtering steps to avoid strand bias.

If replicates are available, we have to consider proper SPCs.
We use JAMM~\citep{Ibrahim2015}, a peak caller that takes replicates into account, to define a peak list and refer to this method as DESeq-JAMM. 
JAMM takes input-DNA into account and subtracts it from ChIP-seq profiles.
Also, we apply IDR~\citep{Li2011} which is a method to define for a set of replicates a list of peaks with high consistency within the replicates.
We follow the framework of ENCODE for the IDR computation (see~\url{https://sites.google.com/site/anshulkundaje/projects/idr}, last access October 14th, 2015).
We refer to this method as DESeq-IDR.


\subsubsection{DBChIP}
The two-stage DPC DBChIP~\citep{liang2012} receives as input the summit (position with maximal count within a peak) information of peaks from SPCs.
The peaks' summits are clustered to obtain consensus peaks.
Then, edgeR~\citep{robinson2010} is applied to derive DPs from the consensus peaks.
If available, input-DNA is subtracted from the ChIP-seq profiles.
DBChIP has as objective only the analysis of transcription factor (TF) peaks and therefore uses predefined short regions of 200 bp around the peak summits as candidates for DPs. 
DBChIP is not able to take replicates into account, to compute the \nuc{GC}-content and to normalize the input-DNA before subtraction.
The fragmentation size can be computed by the SPC that is used.

\subsubsection{MAnorm}
MAnorm~\citep{shao2012} receives as input the candidate peaks from SPCs.
MAnorm normalizes the peak counts between two samples with a local robust regression approach and computes for each candidate peak a $p$-value.
The $p$-value is used to check whether a DP has been found.
The fragmentation size can be computed by the used SPC.
MAnorm is not able to take replicates into account and does not take advantage of input-DNA or \nuc{GC}-content.
Furthermore, no post-precessing steps are performed.

\subsection{One-Stage Differential Peak Caller}
One-stage DPC methods are based on segmentation methods, such as Hidden Markov models (HMMs) or sliding window based approaches. 
While two-stage DPC work in two phases, one-stage DPCs analyze ChIP-seq profiles and perform DP calling in a single step.
We list all one-stage DPC that are available to our knowledge.
Table~\ref{tab_tools} gives an overview of the tools and their supported features.

\subsubsection{ChIPDiff}
To our knowledge, the earliest published method proposed specifically for the differential peak calling problem is ChIPDiff \citep{xu2008}. 
ChIPDiff uses a three state HMM to distinguish between DPs and background signal.
The HMM emission is based on an approximation of a Beta-Binomial distribution, which is fixed after the initialization of the model. 
The Baum-Welch algorithm is used to estimate transition parameters. 
ChIPDiff exhibits some limitations.
Instead of a $p$-value , an empirical fold-change criterion is used to determine whether a DP is significant. 
Moreover, the fragmentation size of a ChIP-seq experiment is fixed to 200bp.
ChIPDiff does not take advantages of input-DNA and does not perform \nuc{GC}-content normalization.
Also, replicates are not supported.


\subsubsection{Csaw} Csaws~\citep{Aaron2014} main method is a based on a window-based approach to segment ChIP-seq profiles.
Replicates can be taken into account.
A modified version of the TMM method is applied to normalize the CHIP-seq signal on 10kbp bins.
EdgeR~\citep{robinson2010}, which is based on a Negative Binomial distribution test, is used to assign a $p$-value to each DP. 
Consecutive significant bins are merged to form final DPs. % followed by a correction of $p$-values following sime's method.
% XXX cite
Input-DNA is not used to normalize ChIP-seq signals, but only in a postprocessing step to filter out potential false positive DPs.
Furthermore, csaw does not normalize against \nuc{GC}-content and does not estimate the fragmentation size. 
% As suggested by the authors, we use a window size of 150bp and a step size of 25bp. 
% All other paprameters are set as default. We were not able to run CSAW on simulated data, even when trying out distinct parameters as used in the real data. 

\subsubsection{PePr} PePr~\citep{Zhang2014} follows a window-based strategy to detect DPs.
The windows size is automatically computed and equals the estimated average width of initially called peaks.
PePr normalize the input-DNA to the mean of all ChIP-seq signals, computes the fold change of input-DNA and ChIP-seq signal and follows the TMM approach to globally normalize across different ChIP-seq profiles. 
PePr requires input-DNA to run.
To check for DPs, first read counts are modelled by a Negative Binomial distribution and second Wald's test is applied to check for significance in read counts. 
Furthermore, PePr provides estimation of fragment size, input subtraction, filtering of peaks with strand bias, but does not correct for \nuc{GC}-content.
PePr can handle replicates of ChIP-seq profiles.

\subsubsection{DiffReps} 
DiffReps~\citep{Li2013} performs a sliding window approach to identify potential DPs. 
It globally normalizes by the geometric mean for each sample and also takes input-DNA into account.
A pre-screening test ensures that only bins with a sufficient number of reads are taken into account.
DiffReps can deal with replicates and uses a Negative Binomial test based on \cite{anders2010} to detect DPs.

\subsubsection{RSEG}
RSEG~\citep{song2011} is specialized for SPCP of repressive histones, which are distributed in very large sequenced genomic domains. 
However, it has an option to call DPs with a three state HMM.
RSEG's HMM uses Difference Negative Binomial distribution as emission distribution.
As it is tailored for broad histone marks, we do not take RSEG into account for this thesis.

\begin{table}[ht]
\begin{center}
 \begin{tabular}{c|c|c|c|c|c|c|c|c|c|c|c}
 & \multicolumn{5}{c|}{characteristics}     & \multicolumn{6}{c}{pre- and postprocessing} \\\hline\hline
  		& \rotatebox{90}{replicates} & \rotatebox{90}{One-Stage DPC} & \rotatebox{90}{Segmentation Strategy} & \rotatebox{90}{statistical model DP} & \rotatebox{90}{peak size} & \rotatebox{90}{frag. size estimation} & \rotatebox{90}{input-DNA norm.} & \rotatebox{90}{Subtracting input-DNA} & \rotatebox{90}{\nuc{GC}-content} & \rotatebox{90}{input-DNA not required } &  \rotatebox{90}{strand bias} \\\hline\hline
 PePr		& $\times$	& $\times$ 	& win		& Wald's test	& s/m & $\times$ 	& $\times$ 	& $\times$ 	& 		& 		& $\times$ \\\hline
 diffReps 	& $\times$	& $\times$ 	& win		& NB 		& s/m & 		& $\times$ 	&  		& 		& $\times$ 	& 		\\\hline
 csaw 		& $\times$	& $\times$ 	& win		& NB 		& s/m & 		& 		&  		& 		& $\times$ 	& 	 	\\\hline
 DiffBind 	& $\times$	&  		& SPC 		& NB 		& s/m & 		& 		& $\times$ 	& 		& $\times$ 	& 		\\\hline
 DESeq-IDR 	& $\times$	&  		& SPC 		& NB 		& s/m & 		& 		& 		& 		& $\times$ 	& 		\\\hline
 DESeq-JAMM 	& $\times$	&  		& SPC 		& GMM, NB	& s/m & $\times$	& 		& $\times$	& 		& $\times$ 	& 		\\\hline\hline
%  ODIN		& 	& $\times$ 	& HMM 		& NB		& $\times$ 	& $\times$ 	& $\times$ 	& $\times$ 	& $\times$ 	& $\times$ \\\hline
 MACS2 		&		& 		& SPC 		& \textit{NA} 	& s/m & $\times$ 	& \textit{NA} 	& \textit{NA} 	& \textit{NA}	& $\times$ 	& \textit{NA}	\\\hline
 DBChIP 	& 		&  		& SPC 		& NB		& s/m & 		& 		& $\times$	& 		& $\times$ 	& 		\\\hline
 MAnorm 	& 		&  		& SPC 		& -		& s/m & 		& 		& 		& 		& $\times$ 	& 		\\\hline
 ChIPDiff	& 		&  $\times$	& HMM 		& -		& s/m & fixed		& 		& 		& 		& $\times$ 	& 		\\\hline
 RSEG 		& 		& $\times$ 	& HMM		& NBDiff 	& l & 		& 		&  		& 		& $\times$ 	& 	 	\\
 \end{tabular}
 \caption[Tool characteristics]{Tool characteristics. 
 Differential peak callers can be categorized in one-stage or two-stage approaches using either an HMM or a window-based approach to segment the ChIP-seq profiles.
 They perform a statistical test based on a Negative Binomial (NB) distribution, a Difference Negative Binomial distribution, Wald's test or Gaussian mixture model (GMM) to identify DPs.
 The tools are specialized in different domain sizes in the ChIP-seq signal. 
 ChIP-seq experiments with small (s) domains are based on TFs, with medium (m) domains on active histone marks, and large~(l) domains on repressive histone marks.
 In this thesis, we investigate differential peak callers that concentrate on small and medium size domains.
 Input-DNA can be normalized and may be used to subtract it from ChIP-seq profiles.
 Also, normalizing against \nuc{GC}-content may prohibit bias in profiles.
 For DESeq-JAMM, JAMM uses GMM to detect peaks and DESeq uses NB to detect DPs. 
 JAMM subtracts the input-DNA from ChIP-seq profiles.
 MAnorm does not model counts of DPs, but normalizes them and assigns directly a $p$-value to them.
}
 \label{tab_tools}
\end{center}
\end{table}
% XXX add DESeq-quest from ODIN paper?





\section{Discussion and Conclusion}
Two challenges naturally arises from ChIP-seq data.
SPCs call peaks on a single ChIP-seq signal and DPCs identify differences in ChIP-seq signals that are associated to two biological conditions.
We divided  the DPCs into two classes: two- and one-stage DPCs.
Two-stage DPCs have clear conceptional disadvantages.
First, their DPs are restricted to their initial candidate regions as well as to the strategy used to create the set of candidate peaks.
These candidate regions depend on the SPC itself as well as the concrete parametrization of the SPC.
Some SPC are specialized in calling broader regions while some SPC show advantages in calling sharp peaks~\citep{wilbanks2010}.
Consequently, prior knowledge of the data is required to obtain accurate peak predictions.
While two-stage DPCS can detect DPs where the number of read counts is significantly higher or lower in one of the ChIP-seq conditions, two-stage DPC fail to detect subtle changes within these candidate regions~\citep{Allhoff2014, Maze2014}. 
This is particularly crucial for ChIP-seq data of histone modifications, where DNA-protein interactions occur in mid size to large domains.
Histone modifications associated to active regulatory regions occur in domains spanning several hundreds of base pairs and may have intricate patterns of gain or loss of ChIP-seq signals within the same domain. 
In contrary, ChIP-seq from transcription factors mostly happens in small isolated peaks.
Figure~\ref{pic_dp_example} shows an example for the predictions of a SCP which are merged by a two-stage DPC to identify DPs.
The two-stage DPC fails to detect DP2, as the SCP predicts a too broad peak in this region such that the signal change of DP2 is not detectable for the DPC.
Furthermore, the SCP calls a domain that contain both DP4 and DP5.
Consequently, it is impossible for the DPC to distinguish between these DPs.
Second, two-stage DPC methods usually do not provide any preprocessing steps crucial for ChIP-seq analysis, such as fragment size estimation, \nuc{GC}-bias correction and input-DNA subtraction (see Section~\ref{sec_chipseq_challenges}). 

The majority of DPCs, namely DiffBind, Csaw, PePr, DiffReps, DESeq and DBChIP, uses TMM (see Section~\ref{sec_back_norm}) or a related approach to normalize ChIP-seq profiles.
However, TMM was devised for gene expression experiments which assumes that counts of most observations (genes or peaks) do not change.
This is not necessarily the case for protein interactions, as two distinct cells can have distinct amounts of proteins or histone modifications bound to their DNA~\citep{Meyer2014}. 
Particularly problematic in this normalization approaches is the effect of replicate specific background noise (see Section~\ref{sec_challenges_norm}). 

Moreover, all methods that solve DPCP with replicates use window-based approaches to identify DPs and apply heuristic strategies to merge peaks (DiffReps, PePr and csaw). 
HMM-based approaches are the more appropriate method to segment a signal, as they intrinsically detect peaks with variable size through the use of posterior decoding algorithms.
Hence, HMMs represent a robust alternative to windows-based segmentation approaches.

There is no method that addresses all pre- and postprocessing steps listed in Table~\ref{tab_tools}.
Only some methods perform preprocessing steps required for the DP analysis.
For instance, only PePr, MACS2 and DESeq-JAMM are able to estimate the fragmentation size of a ChIP-seq experiment.
All other methods, that is, diffReps, csaw, DiffBind, DESeq-IDR, DBChIP, MAnorm and ChIPDiff, need a user defined input for the fragmentation size.
% Because of the complexity of the ChIP-seq protocol, it is recommended to computationally derive the fragmentation size from the data instead of taking the fragmentation size from the protocol~\citep{Pepke2009}.
Further, input-DNA may help to identify technical artifacts and therefore to avoid false positive DPs (see Section~\ref{sec_control_sample}). 
PePr, DiffBind, DESeq-JAMM and DBChIP take advantage of input-DNA, but only PePr also normalizes input-DNA.
No method takes \nuc{GC}-content into account to improve the DP predictions.
Also, only PePr provides postprocessing steps to get rid of implausible DP candidates. 
The main disadvantage of PePr is that is requires input-DNA to predict DP which is not always available.
Furthermore, PePr uses a window-based approach to detect DPs.
No method takes blacklisted genomic regions into account.

The systematic evaluation of DPCs is still an open problem.
An indirect metric, such as the combination of gene expression data with DP estimates, can determine the quality of a DPCP solution.
Moreover, simulated data can help to investigate DPCP solutions in a systematic way as gold standards can be customized.
In particular, it is crucial to have methodologies exploring the performance of DPCs on data with distinct characteristics: from ChIP-seq samples with low variability and high signal-to-noise-ratio to samples with high variability and low signal-to-noise ratios.


\section{Aims of the Thesis}
\label{sec_aims}
In the previous sections, we pointed out that various methods have been developed to solve the differential peak calling problem.
We discussed that two-stage DPCs have conceptual disadvantages and that therefore one-stage DPCs are the favourable method.
Furthermore, there is no method that covers all challenges that have to be considered in the ChIP-seq analysis (see Table~\ref{tab_tools}).
In this thesis we want to propose one-stage DPCs using HMMs that take into account all ChIP-seq associated challenges. 
We restrict our analysis to TF and activating histone marks resulting in relatively small to medium sized peaks in ChIP-seq signals.

We pointed out that normalization of ChIP-seq profiles is a crucial step to identify DPs (see Section~\ref{sec_challenges_norm}).
% Normalization avoids the under- or overrepresentation of ChIP-seq profiles due to different sequencing depths.
% Furthermore, it avoid calling false positive DPs due to different signal-to-noise ratios.
In this thesis, we want to propose a novel normalization strategy that more robust to background noise.
The background noise makes difficulties for TMM or the normalization by library size (see Section~\ref{sec_back_norm}).
Also, ChIP-seq experiments are typically replicated to reduce the effect of technical bias (see Secion~\ref{sec_challenges_rep}).
Hence, our methods have to properly consider overdispersion in their statistical models.

We listed several challenges that either arise from the ChIP-seq protocol itself (see Section~\ref{sec_chipseq_challenges}) or in particular from the differential peak calling problem (see Section~\ref{sec_challenges_dpcp}).
Our proposed methods should address all of them: 
the \nuc{GC}-content (see Section~\ref{sec_challenges_sequencing}), to compensate the correlation between the number of reads and the underlying \nuc{GC}-content;
PCR duplicates (see Section~\ref{sec_challenges_pcr_duplicates}), to avoid signal in ChIP-seq profiles which is based on PCR duplicates rather than biological events;
the input-DNA (see Section~\ref{sec_control_sample}), to get rid of bias in the ChIP-seq data for example due to the shearing process;
the fragment size estimation (see Section~\ref{sec_challenges_shift}), to compute the precise location of the DNA-protein complexes in the genome; and
blacklisted genomic regions (see Section~\ref{sec_challenges_sequencing}), to get rid of DPs that lie within regions that are not properly covered by the sequencing process,

Our methods should take ChIP-seq reads as input that are aligned to a particular genome.
This is a realistic requirement, because in many cases the alignment of reads is either computed in a preprocessing step by specialized software~\citep{li2010} or provided by further sources, for example by public databases~\citep{Edgar2002} or external collaborators.
% Consequently, there is no way to influence the antibody that is used in the ChIP-seq experiment, the number of reads and the number of uniquely mappable reads (see Section~\ref{sec_challenges_sequencing}).

A further aim of this thesis is to develop evaluation strategies for differential peak calling solutions.
As described in Section~\ref{sec_challenges_eval} there is neither a gold standard nor a direct metric to check the quality of a differential peak calling solution.
% There is a clear need for a metric to systematically evaluate DPCs.
% Hence, a further aim of this thesis is the development of such evaluation strategies. 
Evaluation strategies can assess DP estimates in a systematic way.
% The aim is to rank DP predictions estimated by various methodologies by a metric.
% Such a ranking can then serve as base to derive the most competing tools from it.
% By using data sets with various characteristics, such as the sequencing depth or consistency among replicates, it is possible to decipher the advantages and disadvantages of certain DPCs. 
% Thereby, the tool's strengths as well as weak points can be discovered.
% Moreover, simulation of ChIP-seq signals may help to rate the peak predictions of DPCs.
In this thesis, we want to propose an indirect metric to quantify DPCs.
Moreover, we should develop a simulation algorithm to be able to produce customized gold standards.
With regard to these evaluation strategies our methods should give best results.
% In particular, our method should be robust against data characteristics such as the sequencing depth or the level of overdispersion in the data.







